src/coupling/hc_explicit_sequential.cc

```diff
@@ -175,7 +175,7 @@ void HC_ExplicitSequential::advection_process_step(AdvectionData &pdata)
         // for simplicity we use only last velocity field
         if (pdata.velocity_changed) {
             //DBGMSG("velocity update\n");
-            pdata.process->set_velocity_field( water->get_mh_dofhandler() );
+            pdata.process->set_velocity_field( std::dynamic_pointer_cast<DarcyMH>(water)->get_velocity_field(), water->get_mh_dofhandler() );
```

Comment:
- remove passing mh_dofhandler
- make velocity part of the equation's filed set and replace get_velocity_field with FieldSet::field (should be changed to return shared pointer)

Reply:
First is solved.
But FV transport needs velocity_last_time (last time of DarcyMH, mh_dh->time_changed() in master. We must remove passing of this time and then we can solve second point.

Reply:
FV Transport should use the times of the velocity field. Usuall procedure is:

```
changed = velocity_field.set_time(current_time);
if (changed) {
	assembly_matrix();
}
```

Compare it with DG transport. 
However, this implies changes in the logic of FV transport updates which must be done very carefully.

Please, make a new issue for this and copy there this conversation.

Reply:
Will be solved in #1085